CRM/Contact/Form/CustomData.php

```diff
@@ -85,11 +85,10 @@ class CRM_Contact_Form_CustomData extends CRM_Core_Form {
   public function preProcess() {
     $this->_cdType = $_GET['type'] ?? NULL;
     $this->assign('cdType', FALSE);
+    $isBuildForm = ($_GET['type'] ?? NULL) && CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this);
     $this->_multiRecordDisplay = CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this);
-    if ($this->_cdType || $this->_multiRecordDisplay == 'single') {
-      if ($this->_cdType) {
-        $this->assign('cdType', TRUE);
-      }
+    $this->assign('cdType', CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this));
```

Comment:
It looks like 'cdType' was already assigned on L87 and you're re-assigning here?

Reply:
@colemanw yeah - so my read is the first one can go?

Reply:
Looks that way

Reply:
@colemanw OK I fixed that - I pushed in 2 more minor fixes - code to set an unused variable & also this business of passing `$entityId` into a function that makes it a form property & then using the form property

![image](https://github.com/civicrm/civicrm-core/assets/336308/1966ec2a-5454-4e78-b984-913796396e3f)